detailed_docs/Wiki/0249_capabilities.md

```diff
@@ -0,0 +1,135 @@
+
+**Given**  Receiving HMI response for any of the UI/VR/TTS/Buttons/VehicleInfo/Navigation/RC capabilities
+
+**When** SDL checks if `HMICapabilitiesCacheFile` file already has the data set received in HMI response` and `HMICapabilitiesCacheFile` file does not have the data set received in HMI response
+
+**Then** SDL writes the HMI response data to `HMICapabilitiesCacheFile` file  
+**And** SDL does not send corresponding requests to get HMI capabilities to HMI for subsequent ignition cycles, follow the rest of launch sequence requests/notifications as applicable.  
+**And** SDL uses this new data set for apps/SDL Core communication until ignition cycle is performed.
+
+---
+
+#### Scenario 5:  SDL continues launching sequence if capabilities sent fom HMI are already present in `HMICapabilitiesCacheFile` file
+
+**Given**  Receiving HMI response for any of the UI/VR/TTS/Buttons/VehicleInfo/Navigation/RC capabilities
+
+**When** `HMICapabilitiesCacheFile` file already has the data set received in HMI response
+
+**Then** SDL does not overwrite the response to `HMICapabilitiesCacheFile` file  
+**And** SDL does not send corresponding request to get HMI capabilities to HMI for subsequent ignition cycles  
+**And** SDL follows the rest of the launch sequence
+
+---
+
+#### Scenario 6: SDL overwrites contents of received Dynamic RPC in hmi_capabilities_cache.json file
+
+**Given** Mobile application already obtained HMI capabilities in the ignition cycle
+
+**When** HMI sends a Dynamic RPC with changes to capabilities
+
+**Then** Core must overwrite contents of received notifications in `HMICapabilitiesCacheFile` file and use this data until ignition cycle is performed
+
+**_Dynamic RPCs:_**
+  * UI.OnLanguageChange
+  * VR.OnLanguageChange
+  * TTS.OnLanguageChange
+
+---
+
+#### Scenario 7: SDL deletes `HMICapabilitiesCacheFile` file when a master reset is performed
+
+**Given** All HMI Capabilities are presented in `HMICapabilitiesCacheFile` 
+
+**When** HMI sends OnExitAllApplications with reason `MASTER_RESET`
+
+**Then** SDL Core must delete `HMICapabilitiesCacheFile` in AppStorageFolder in [smartDeviceLink.ini](https://github.com/smartdevicelink/sdl_core/blob/master/src/appMain/smartDeviceLink.ini)
+
+## Feature 2: As a mobile application, I want to get actual HMI capabilities from SDL after HMI SW update
+
+### Scenario 1: SDL regenerates `HMICapabilitiesCacheFile` file when system SW version changes
+
+**Given** App sends RAI after HMI SW update 
+
+**When** SDL's local `ccpu_version` does not match with received from BC.GetSystemInfo(`ccpu_version`)
+
+**Then** SDL must delete `HMICapabilitiesCacheFile`  
+**And** SDL must send `GetCapabilities` requests for each interface (VR, TTS, UI, etc) to HMI and persist the responses received from HMI to `HMICapabilitiesCacheFile`  
+**And** SDL must set `IsHMICooperating` to true when HMI has responded to all the HMI capabilities requests  
```

Comment:
@KhrystynaDubovyk it is an internal SDL variable and I don't think it is testable